go-controller/pkg/factory/factory.go

```diff
@@ -26,16 +26,22 @@ type informer struct {
 }
 
 func (i *informer) forEachHandler(obj interface{}, f func(id uint64, handler cache.ResourceEventHandler)) {
+	handlersCopy := make(map[uint64]cache.ResourceEventHandler)
+
 	i.Lock()
-	defer i.Unlock()
+	// copying the informer handlers and releasing the lock prevents deadlocks
```

Comment:
can you add  "prevents deadlocks when trying to change handlers within a handler"?